server/api/availability.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package api
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/job"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/bot"
+)
+
+const (
+	AVAILABILITY_VIEW_FREE              = '0'
+	AVAILABILITY_VIEW_TENTATIVE         = '1'
+	AVAILABILITY_VIEW_BUSY              = '2'
+	AVAILABILITY_VIEW_OUT_OF_OFFICE     = '3'
+	AVAILABILITY_VIEW_WORKING_ELSEWHERE = '4'
+)
+
+type availabilityJob struct {
+	api API
```

Comment:
I tried to avoid this pattern in `notification.go` - the thinking was that `api` is for an "acting user", has (at least) the current user's MattermostUserID etc. associated with it.

As I was working on solar lottery, i realized that this totally makes sense, because the `api` should just be operating in the context of the bot user. In fact, I am thinking that maybe the Bot User is the right concept to use for the "application-level" client. Can talk offline about it, no immediate change request here.